File: tests/fixtures/lang_samples/cpp/nested.cpp
// Fixtures for nesting / cyclomatic-complexity walker tests.

int deeplyNested(int x) {
    if (x > 0) {
        for (int i = 0; i < x; i++) {
            while (i > 0) {
                if (i == 5) {
                    return i;
                }
            }
        }
    }
    return x;
}

int shallow(int x) {
    return x + 1;
}

int manyBranches(int x) {
    if (x == 1) return 1;
    if (x == 2 && x > 0) return 2;
    if (x == 3 || x < 0) return 3;
    if (x == 4) return 4;
    if (x == 5) return 5;
    if (x == 6) return 6;
    return x;
}

File: tests/fixtures/lang_samples/scala/nested.scala
// Fixtures for control-flow walker tests (nesting, CCN, guards, match).

object Nested {
  def deeplyNested(items: List[Int]): Int = {
    var total = 0
    for (a <- items) {
      if (a > 0) {
        for (b <- items) {
          if (b > 1) {
            while (total < 10) {
              total += 1
            }
          }
        }
      }
    }
    total
  }

  def manyBranches(a: Int, b: Int): Int = {
    if (a == 1) { return 1 }
    else if (a == 2) { return 2 }
    else if (a == 3) { return 3 }
    else if (a == 4) { return 4 }
    else if (a == 5) { return 5 }
    if (a > 10 && b > 0 || b < -5) { return 6 }
    0
  }

  def matchGuards(n: Int): String = {
    n match {
      case 0 => "zero"
      case x if x > 100 => "big"
      case x if x < -100 => "small"
      case _ => "mid"
    }
  }

  def flatMatch(n: Int): String = {
    n match {
      case 0 => "zero"
      case 1 => "one"
      case _ => "other"
    }
  }

  def forGuard(items: List[Int]): Int = {
    var n = 0
    for (i <- items if i > 0) {
      n += i
    }
    n
  }

  def shallow(x: Int): Int = x + 1
}
